Minor maintenance things

Zanthed opened the pull request:
Signed-off-by: June <aobugene@reliaquest.com>

lberrymage commented on README.md:
```diff
@@ -23,7 +23,7 @@ on macOS.
 ## Dependencies
 
-You will need the following dependencies if you are compiling from Linux:
+If you are compiling on Linux, make sure to install the dependencies below.
```
Can you either squash this commit into your first one or remove it? I'm okay with the wording either way, so it's up to you.

lberrymage commented on README.md:
```diff
@@ -21,7 +21,39 @@ cargo run --features "metal"
 
 on macOS.
 
-### Issues
```
I would prefer that this stays under the "How to run" section since it directly relates there. I do appreciate the alphabetization though :grin:.

lberrymage commented on README.md:
```diff
@@ -21,7 +21,39 @@ cargo run --features "metal"
+
+### Other
+
+See your distribution-specific installation process for the equivalent dependencies.
```
This is a bit of a nitpick, but could you wrap this at 80 characters? Thanks.

aayux replied:
Got it! Only this line or also the install instructions?

lberrymage replied:
I think only this line. The install commands should be on one line for copy-pasting anyway.

lberrymage commented on README.md:
```diff
@@ -46,4 +80,4 @@ for inclusion in the work by you, as defined in the Apache-2.0 license, shall be
 dual licensed as above, without any additional terms or conditions.
 
 
-[Amethyst game engine]: https://amethyst.rs
+[Amethyst game engine]: https://amethyst.rs
```
Missing newline at EOF. An `echo "" >> README.md` should do the trick.

aayux replied:
Ah, my bad.

aayux replied:
Ah, my bad. Should be fixed now.

lberrymage commented:
bors r+

Commit:
Remove empty XML tags

Signed-off-by: June <aobugene@reliaquest.com>

Commit:
Always show an exception message to user

Helps with troubleshooting to avoid the user
needing ADB.

Signed-off-by: June <aobugene@reliaquest.com>

lberrymage committed:
Don't use glob/wildcard imports

They're not really good practice and can lead to future confusion about
where things are coming from.

Co-authored-by: June <aobugene@reliaquest.com>

lberrymage commented on app/build.gradle.kts:
```diff
@@ -75,6 +75,7 @@ android {
     }
     buildFeatures {
         compose = true
+        viewBinding = true
```
I'd rather not enable this until we use it since it might increase compile times (even if only marginally).

lberrymage commented on app/src/main/java/app/accrescent/client/ui/MainActivity.kt:
```diff
@@ -63,7 +72,7 @@ fun MainContent() {
                 "app_details/{appId}",
                 deepLinks = listOf(navDeepLink {
-                    uriPattern = "https://accrescent.app/app/{appId}"
+                    uriPattern = "https://${RepoDataFetcherImpl.ROOT_DOMAIN}/app/{appId}"
```
Could you move `ROOT_DOMAIN` to `data/Urls.kt` or something so we don't need to pull it from `RepoDataFetcherImpl`?

lberrymage commented on app/src/main/java/app/accrescent/client/Accrescent.kt:
```diff
@@ -24,8 +29,8 @@ class Accrescent : Application(), Configuration.Provider {
-                "User action required",
-                NotificationManager.IMPORTANCE_DEFAULT
+                getString(R.string.user_action_required),
+                NotificationManager.IMPORTANCE_HIGH
```
I'm not sure how I feel about raising the notification importance. `IMPORTANCE_HIGH` is too intrusive to me personally so I want to get users' feedback before changing it. For precedence, GrapheneOS [implicitly uses `IMPORTANCE_DEFAULT`](https://github.com/GrapheneOS/platform_packages_apps_GmsCompat/blob/90222fd16faf2753a0c7437dce0b0fb2ed4bf939/src/app/grapheneos/gmscompat/BinderGms2Gca.kt#L77) in its Play Store hooks.

lberrymage commented on app/src/main/java/app/accrescent/client/util/ApkDownloader.kt:
```diff
@@ -7,6 +7,8 @@ import android.content.res.Resources
 import android.util.DisplayMetrics
 import android.util.Log
+import app.accrescent.client.R
+import app.accrescent.client.data.RepoDataFetcherImpl.Companion.REPOSITORY_URL
```
Same as above. Move `REPOSITORY_URL` to another file (preferably the same one as `ROOT_DOMAIN`) if we're using it here.

lberrymage commented on app/src/main/java/app/accrescent/client/util/ApkDownloader.kt:
```diff
@@ -94,15 +96,16 @@ class ApkDownloader @Inject constructor(
                 screenDensity <= DisplayMetrics.DENSITY_XXHIGH -> "xxhdpi"
-                else -> "xxxhdpi"
+                screenDensity <= DisplayMetrics.DENSITY_XXXHIGH -> "xxxhdpi"
+                else -> "nodpi"
```
If `screenDensity` is larger than `DENSITY_XXHIGH` I think `xxxhdpi` is the correct fallback. Low-DPI devices won't ever reach this point so they won't incorrectly fall back to `xxxhdpi`.

lberrymage commented on app/src/main/res/values/strings.xml:
```diff
@@ -2,4 +2,21 @@
+    <string name="version_downgrade">Version %1$d is less than minimum %2$d</string>
+    <string name="app_id_mismatch">App ID %1$s does not match expected ID %2$s</string>
+    <string name="version_mismatch">Expected version %1$s, actual is %2$s</string>
+    <string name="version_code_mismatch">Expected version code %1$d, actual is %2$d</string>
```
nit: Keep error messages starting with a lowercase letter. They're always displayed with context and not capitalizing them makes them easier to display in arbitrary places.

lberrymage commented on app/src/main/res/values/strings.xml:
```diff
@@ -2,4 +2,21 @@
+    <string name="version_code">Version code: %1$d</string>
+    <string name="device_language_unsupported">Your device\'s language is not supported</string>
+    <string name="device_density_unsupported">Your device\'s screen density is not supported</string>
+    <string name="device_abi_unsupported">Your device\'s ABI/architecture is not supported</string>
```
nit: Say either "ABIs are" or "architecture is" instead of both. Singular "ABI" isn't always technically correct since devices can support multiple ABIs.

lberrymage commented on app/src/main/java/app/accrescent/client/util/ApkDownloader.kt:
```diff
@@ -46,23 +48,23 @@ class ApkDownloader @Inject constructor(
             ?: throw InvalidObjectException("base.apk is not a valid APK")
```
You can use `R.string.base_apk_not_valid` here.

lberrymage committed:
Move REPOSITORY_URL to separate file & add ROOT_DOMAIN constant

Co-authored-by: June <aobugene@reliaquest.com>

lberrymage committed:
Migrate some strings to resources where possible

Co-authored-by: June <aobugene@reliaquest.com>
